src/program/mod.rs

```diff
@@ -62,11 +62,57 @@ lazy_static! {
     static ref COMPILER_GLOBAL_LOCK: Mutex<()> = Mutex::new(());
 }
 
+/// Used in ProgramCreationError::CompilationError to explain which shader stage failed compilation 
+#[derive(Copy, Clone, Debug)]
```

Comment:
Maybe also derive Eq?

Reply:
I'll fix.

Reply:
I'll also make sure to rebase and get rid of the merge commit. I had attempted to do so, but obviously made some sort of mistake.